common/partition/interface.go

```diff
@@ -30,15 +30,17 @@ import (
 	"github.com/uber/cadence/common/dynamicconfig"
 )
 
-// Config is the base configuration for the partitioning library
-type Config struct {
-	IsolationGroupEnabled dynamicconfig.BoolPropertyFnWithDomainFilter
-	AllIsolationGroups    []string
-}
-
 type Partitioner interface {
 	// GetIsolationGroupByDomainID gets where the task workflow should be executing. Largely used by Matching
 	// when determining which isolationGroup to place the tasks in.
 	// Implementations ought to return (nil, nil) for when the feature is not enabled.
 	GetIsolationGroupByDomainID(ctx context.Context, DomainID string, partitionKey PartitionConfig) (*string, error)
 }
+
+// Config is the configuration for the partitioning library, a library
+// for segmenting portions of workflows into isolation-groups - a resiliency
+// concept meant to help move workflows around and away from failure zones.
+type Config struct {
```

Comment:
It can be removed

Reply:
I would prefer to keep given configuration tends to evolve and change and having a wrapper struct makes for a much easier set of changes in the future